runtime/include/jvmti.h.m4

```diff
@@ -47,8 +47,9 @@ JNIEXPORT jint JNICALL Agent_OnAttach(JavaVM* vm, char *options, void *reserved)
 #define JVMTI_VERSION_1 (JVMTI_VERSION_1_0)
 #define JVMTI_VERSION_9  0x30090000
 #define JVMTI_VERSION_11 0x300b0000
+#define JVMTI_VERSION_15 0x300f0000
 
-#define JVMTI_VERSION JVMTI_VERSION_11
+ifelse(eval(JAVA_SPEC_VERSION >= 15), 1, [#define JVMTI_VERSION JVMTI_VERSION_15], [#define JVMTI_VERSION JVMTI_VERSION_11])
```

Comment:
Does this mean that Java 8 is incorrect?

Reply:
It doesn't match Oracle - there is no java 8 specific version, it was still 1.something at that point, I think.

Reply:
`JVMTI_VERSION` does exist in Oracle JDK8.

Reply:
`JDK8` didn't need special handling and worked with `#define JVMTI_VERSION JVMTI_VERSION_11`. I think `JDK8 JCL jdwp` native doesn't use `JVMTI_VERSION`.

Reply:
I think it needs to be fixed for Java 8, it should be 1.2.1.
Seems OpenJ9 Java 8 builds use OpenJDK jvmti.h, so it's correct there, but other builds (uhm IBM) have the wrong value.

Reply:
> I think it needs to be fixed for Java 8, it should be 1.2.1.

Is it to correct the `jvmti.h` file only? I assume there is no functional difference for IBM JDK8 itself since current `JVMTI_VERSION` value has been defined since it was introduced by `JDK11`.

Reply:
Others may be lazy in their agents and use the constant (this is extremely foolish, since you have to know what verison of TI you need to get any use out of it - you'd have to be using the lowest possible version of TI for this  to work at all).

Reply:
> Is it to correct the jvmti.h file only?

Yes, it's clearly wrong for Java 8.  Anything using that constant won't necessarily run on Java 8. Agreed it won't make any functional difference to Java 8. Not sure why the IBM testing doesn't have any issue.

Reply:
Currently `OpenJ9 JDK8` returns `JVMTI_1_2_3_SPEC_VERSION (1.2.3)`. 
https://github.com/eclipse/openj9/blob/d89764f909755360950848a474c75c3ba8bb791f/runtime/jvmti/jvmtiGeneral.c#L182 
Should it be changed to `JVMTI_1_2_SPEC_VERSION (1.2.1)` to match RI instead?

Reply:
Seems a good idea to match the RI. This can't have much field impact, I note it was version 9 since 2016 before it changed to version 11.

Reply:
Sorry, I have to change the request. Initially I saw following defines in RI `jvmti.h`
```
enum {
    JVMTI_VERSION_1   = 0x30010000,
    JVMTI_VERSION_1_0 = 0x30010000,
    JVMTI_VERSION_1_1 = 0x30010100,
    JVMTI_VERSION_1_2 = 0x30010200,

    JVMTI_VERSION = 0x30000000 + (1 * 0x10000) + (2 * 0x100) + 1  /* version: 1.2.1 */
};
```
It is clear that `JVMTI_VERSION` is `1.2.1` with value of `0x30010201`. However I run this same JVM to against a simple testcase invoking `jvmti->GetVersionNumber(version_ptr)`, and got `0x30010203` which is `1.2.3`, i.e., `JVMTI_1_2_3_SPEC_VERSION` currently returned by `OpenJ9`, so this was to match RI behaviour, and should keep that way instead.

It appears there is an internal define within RI.

Reply:
Limited `JVMTI_VERSION_15` for `JDK15`, `JVMTI_VERSION_11` for `JDK11+`,
`JVMTI_VERSION_9` for `JDK9+`, also defined `JVMTI_VERSION` accordingly.
Please re-review.